plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveBucketing.java

```diff
@@ -373,6 +364,18 @@ private static boolean isTypeSupportedForBucketing(TypeInfo type)
         throw new UnsupportedOperationException("Unknown type " + type);
     }
 
+    private static boolean isTypeSupportedForBucketFiltering(TypeInfo type)
```

Comment:
We also have a `isTypeSupportedForBucketing` in this class.
It seems weird that a type would be supported for bucketing but not for bucket filtering as there wouldn't be much point to bucketing by something if we can't also use it for filtering.
@findepi @Praveen2112 any idea why this might be the case ?

Reply:
I've tried this locally and it should work unless there's a edge case which I'm missing.

Reply:
I've updated the pr to all supported types